src/AppBundle/Controller/MailingListController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Semester;
+use InvalidArgumentException;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use AppBundle\Form\Type\GenerateMailingListType;
+use Symfony\Component\HttpFoundation\Request;
+
+class MailingListController extends Controller
+{
+    /**
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showAction(Request $request)
+    {
+        $semesters = $this->getDoctrine()->getRepository('AppBundle:Semester')->findAllSemesters();
+
+        $form = $this->createForm(new GenerateMailingListType($semesters));
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $data = $form->getData();
+            $type = $data['type'];
+            $semesterID = $data['semester']->getId();
+
+            switch ($type) {
+                case 'Assistent':
+                    return $this->redirectToRoute('generate_assistant_mail_list', array('semesterID' => $semesterID));
+                case 'Team':
+                    return $this->redirectToRoute('generate_team_mail_list', array('semesterID' => $semesterID));
+                case 'Alle':
+                    return $this->redirectToRoute('generate_all_mail_list', array('semesterID' => $semesterID));
+                default:
+                    throw new InvalidArgumentException('type can only be "Assistent", "Team" or "Alle". Was: '.$type);
+            }
+        }
+
+        return $this->render('mailing_list/generate_mail_list.html.twig', array(
+            'form' => $form->createView(),
+        ));
+    }
+
+    /**
+     * @param int $semesterID
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showAssistantsAction(int $semesterID)
```

Comment:
Tips: Hvis du endrer parameternavn i routing fra `semesterID` til `id`: 
```
path: /kontrollpanel/epostlister/assistenter/{id}
```
kan du finne semesteret direkte 
```
public function showAssistantsAction(Semester $semester)
```

Dette gjelder også for de to andre action-funksjonene